### PR TITLE
Created IMotor

### PR DESCRIPTION
**Pull request by renatodellosso**

### CODE REVIEW

File: RobotCode/src/main/java/frc/lib/core/motors/TeamSparkMAX.java [hunk 1]
```diff
@@ -153,4 +153,69 @@ public void set(final double power, final String reason)
...
+    CanPidController.setOutputRange(-PEAK_OUTPUT, PEAK_OUTPUT);
+  }
+
+  @Override
+  public void setClosedLoopErrorLimit(final double ERROR_TOLERANCE, final int SLOT)
+  {
+    CanPidController.setSmartMotionAllowedClosedLoopError(ERROR_TOLERANCE, SLOT);
+  }
+
+  public void onPidPeriodic(final int SLOT)
```
Comment (miruaraya) [inline RobotCode/src/main/java/frc/lib/core/motors/TeamSparkMAX.java:193]: if overriding this method, should signify

File: RobotCode/src/main/java/frc/lib/core/motors/TeamTalonFX.java [hunk 2]
```diff
@@ -161,4 +161,42 @@ public void configureWithPidParameters(final PidParameters pidParameters, final
...
+    configPeakOutputForward(PEAK_OUTPUT, 30);
+    configPeakOutputReverse(-PEAK_OUTPUT, 30);
+  }
+
+  @Override
+  public void setClosedLoopErrorLimit(final double ERROR_TOLERANCE, final int SLOT)
+  {
+    configAllowableClosedloopError(SLOT, (int) ERROR_TOLERANCE, 30);
+  }
+
```
Comment (miruaraya) [inline RobotCode/src/main/java/frc/lib/core/motors/TeamTalonFX.java:201]: no onPidPeriodic() body for this class, I don't know if that is something you would want/need to add

Comment (renatodellosso) [reply RobotCode/src/main/java/frc/lib/core/motors/TeamTalonFX.java:201]: `onPidPeriodic` has a body in IMotor, so it doesn't have to have a body in implementations.

**Review by miruaraya — APPROVED** [review]
Looks good

**Pull request merged**